core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -186,4 +224,43 @@ ManifestFile manifest() {
       return manifest;
     }
   }
+
+  static class ContentFileStructWithMetrics implements StructLike {
+    private final int structSize;
+    private final StructLike fileAsStruct;
+    private final MetricsUtil.ReadableMetricsStruct readableMetrics;
+
+    ContentFileStructWithMetrics(
+        int structSize,
+        StructLike fileAsStruct,
+        MetricsUtil.ReadableMetricsStruct readableMetrics) {
+      this.structSize = structSize;
```

Comment:
Since we only use structSize -1, why not instead store (and change variable name) to
```
lastStructIndex = fileAsStruct.size() - 1;
```

Reply:
I do use the variable in size().  I changed to use a local variable in get() that is fileAsStruct.size() - 1, hope that works.